joomla/joomla-cms
# Improvements to article modals (with Create Article button)

Pull Request for Improvement.
### Summary of Changes

This PR intends to do several things in the article modal field:
- The most important, add a new button to create a new article directly (ex: when creating a menu item single article type)

This can be very useful since, with this, you can create a New article directly when you are creating a menu item. Save a lot of work when you just need to do this.

Also:
- Add the ability to choose whether to show or not the edit, create, select, or clear buttons.
- Change the buttons behaviour. When no article is selected you have "Select" and "Create", when an article is selected you have "Edit" and "Clear".
- Simplify js in modals by moving most js to a new js file. Instead of inline js.
- Allow to change the category and language in edit modals (this was blocked everywhere because of associations). Now change is allowed, except in associations where the language is blocked and the category only shows the categories for the forced language and All language.
- When the language is forced (associations) now it only shows the tags for that language or the all language
- When the language is forced (associations) now it only shows the categories for that language or the all language

Similiar PR:
- Article: https://github.com/joomla/joomla-cms/pull/11830 (this one)
- Category: https://github.com/joomla/joomla-cms/pull/11857
- Contact: https://github.com/joomla/joomla-cms/pull/11858
- Newsfeed: https://github.com/joomla/joomla-cms/pull/11862
### Animated Gif (click to view in full screen)

![bb](https://cloud.githubusercontent.com/assets/9630530/18105791/b9af4d4a-6ef8-11e6-8e1e-ab568c8b3a65.gif)
### Testing Instructions
1. Use latest staging and apply this patch
2. All article modal fields should work (menu item single article, user profile plugin tos, tinymce select article, article associations, other?).
3. Test all the "Edit", "Create", "Select" and "Clear" buttons.
4. Test Associations modals because of the forcedLanguage.
5. Try also to add several modals in the same page and check all work fine.
6. Code review to check all is ok.
### Documentation Changes Required

Don't know if exist any related to modal fields.
### B/C

I tried to make this all B/C (didn't remove now obsolete code in article edit.php and modal.php).
Anyway, not 100% sure if B/C, so please check if i didn't forget anything.
